public-to-private/run-workflow/main.go

```diff
@@ -0,0 +1,229 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-github-actions/public-to-private/run-workflow/action"
+
+	"github.com/google/go-github/v47/github"
+	"github.com/sethvargo/go-githubactions"
+	"golang.org/x/oauth2"
+)
+
+const COMPLETED_STATUS = "completed"
+
+// ConnectToGithub Sets up the github client and connects
+func ConnectToGithub(githubToken string) *github.Client {
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: githubToken},
+	)
+	tc := oauth2.NewClient(context.Background(), ts)
+	return github.NewClient(tc)
+}
+
+func StartWorkflow(client *github.Client, inputs *action.Inputs) error {
+	fmt.Printf("Starting workflow %s\n", inputs.WorkflowFile)
+	inputsMap, err := inputs.InputsToMap()
+	if err != nil {
+		return err
+	}
+	event := &github.CreateWorkflowDispatchEventRequest{
+		Ref:    inputs.Branch,
+		Inputs: inputsMap,
+	}
+	resp, err := client.Actions.CreateWorkflowDispatchEventByFileName(context.Background(), inputs.Owner, inputs.Repository, inputs.WorkflowFile, *event)
+
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != 204 {
+		return fmt.Errorf("Failed to start the workflow, status code: %d", resp.StatusCode)
+	}
+
+	return nil
+}
+
+func GetListOfWorkflowRuns(client *github.Client, inputs *action.Inputs, workflowStartedAt time.Time) (*github.WorkflowRuns, error) {
+	opts := &github.ListWorkflowRunsOptions{
+		Branch:  inputs.Branch,
+		Event:   "workflow_dispatch",
+		Created: fmt.Sprintf(">=%s", workflowStartedAt.Format(time.RFC3339)),
+	}
+	runs, resp, err := client.Actions.ListWorkflowRunsByFileName(context.Background(), inputs.Owner, inputs.Repository, inputs.WorkflowFile, opts)
```

Comment:
I don't think these are applicable:
1. We limit to just workflow_dispatch events so it does not matter what other triggers/events have happened.
2. Even without a concurrency of 1 we are limiting to the branch, time after we started, event, and actor, and on top of this we grab the latest run to kick off. The only different job we could get is if a new commit to the pr is pushed extremely quickly after one that started the workflow. At which point we don't care about this workflow anymore and will only care about the workflow running from the new commit.

Now along these lines it might be nice to later add in the ability to cancel the private workflow if this one is cancelled to save some on github costs.